Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -722,6 +730,96 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Resolve Conflict SG", "[.SyncServe
     }
 }
 
+TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Auto Purge Enabled - Filter Revoked Revision - SGColl", "[.SyncServerCollection]") {
+     // Admin user for REST requests
+    const string docIDstr = timePrefix() + "apefrr-doc1";
+
+    HTTPStatus status;
+    C4Error error;
+    sendRemoteRequest("POST", "_user", &status, &error, R"({"name":"filterRevoked","password":"password"})", true);
+    sendRemoteRequest("PUT", "_user/filterRevoked", &status, &error, R"({"admin_channels":["*"]})", true);
+    REQUIRE(status == HTTPStatus::OK);
+
+    // Setup pull filter to filter the _removed rev:
+    _pullFilter = [](C4CollectionSpec collectionSpec, C4String docID, C4String revID,
+                     C4RevisionFlags flags, FLDict flbody, void *context) {
+        if ((flags & kRevPurged) == kRevPurged) {
+            ((ReplicatorAPITest*)context)->_counter++;
+            Dict body(flbody);
+            CHECK(body.count() == 0);
+            return false;
+        }
+        return true;
+    };
+
+    // Set up replication parameters
+    constexpr size_t collectionCount = 1;
+    std::array<C4CollectionSpec, collectionCount> collectionSpecs = {
+            Roses
+    };
+    std::array<C4Collection*, collectionCount> collections =
+            collectionPreamble(collectionSpecs, "filterRevoked", "password");
+    std::array<C4ReplicationCollection, collectionCount> replCollections {
+        {{
+            collectionSpecs[0], kC4Disabled, kC4OneShot,
+            nullslice, nullptr, _pullFilter, this
+        }}
+    };
+    C4ParamsSetter paramsSetter = [&replCollections](C4ReplicatorParameters& c4Params) {
+        c4Params.collectionCount = replCollections.size();
+        c4Params.collections = replCollections.data();
+    };
+
+    // Setup onDocsEnded:
+    _enableDocProgressNotifications = true;
+    _onDocsEnded = [](C4Replicator* repl,
+                      bool pushing,
+                      size_t numDocs,
+                      const C4DocumentEnded* docs[],
+                      void* context) {
+        for (size_t i = 0; i < numDocs; ++i) {
+            auto doc = docs[i];
+            if ((doc->flags & kRevPurged) == kRevPurged) {
+                ((ReplicatorAPITest*)context)->_docsEnded++;
+            }
+        }
+    };
+
+    sendRemoteRequest("PUT", docIDstr, &status, &error,
+                      "{\"channels\":[\"a\"],\"scopes\":{\"flowers\":{\"collections\":{\"roses\":{}}}}}"_sl);
+    REQUIRE(status == HTTPStatus::Created);
```

Comment:
I don't know whether the "scope" property will make SG to put the doc to the desired collection. If not, please use the overload that takes collection path as a paramter.

Reply:
In order to do that would I need to merge/rebase master in order to access that function?